Scripts/BuildPhases/LintAppLocalizedStringsUsage.swift

```diff
@@ -115,135 +129,115 @@ extension Xcodeproj {
             guard let obj = wrapped.wrappedValue as? T else {
-                throw DecodingError.unexpectedObjectType(id: id, expectedType: T.self, found: wrapped.wrappedValue)
+                throw ProjectInconsistencyError.unexpectedObjectType(id: id, expectedType: T.self, found: wrapped.wrappedValue)
             }
             return obj
         }
-
+        
         /// Returns the `PBXObject` instance with the given `ObjectUUID`, by looking it up in the list of `objects` registered in the project.
         func object<T: PBXObject>(id: ObjectUUID) -> T? {
             try? object(id: id) as T
         }
-
+        
         /// The `PBXProject` corresponding to the `rootObject` of the project file.
         var rootProject: PBXProject { try! object(id: rootObject) }
-
+        
         /// The `PBXGroup` corresponding to the main groop serving as root for the whole hierarchy of files and groups in the project.
         var mainGroup: PBXGroup { try! object(id: rootProject.mainGroup) }
-
+        
         /// The list of `PBXNativeTarget` targets found in the project.
         var nativeTargets: [PBXNativeTarget] { rootProject.targets.compactMap(object(id:)) }
-
+        
         /// The list of `PBXBuildFile` build file references included in a given target.
         func buildFiles(for target: PBXNativeTarget) -> [PBXBuildFile] {
             guard let sourceBuildPhase: PBXSourcesBuildPhase = target.buildPhases.lazy.compactMap(object(id:)).first else { return [] }
             return sourceBuildPhase.files.compactMap(object(id:)) as [PBXBuildFile]
         }
     }
-
-    /// Helper type to ensure the `isa` field of a `PBXObject` contains the name of the expected type to decode as its value.
-    struct ISA<T>: RawRepresentable, Decodable, CustomDebugStringConvertible {
-        var rawValue: String
-        init?(rawValue: String) {
-            guard rawValue == String(describing: T.self) else { return nil }
-            self.rawValue = rawValue
-        }
-        var debugDescription: String { self.rawValue }
-    }
-
+    
     /// One of the many `PBXObject` types encountered in the `.pbxproj` file format.
     /// Represents the root project object.
     struct PBXProject: PBXObject {
-        private let isa: ISA<Self>
-
         let mainGroup: ObjectUUID
         let targets: [ObjectUUID]
     }
-
+    
     /// One of the many `PBXObject` types encountered in the `.pbxproj` file format.
     /// Represents a native target (i.e. a target building an app, app extension, bundle...).
     /// - note: Does not represent other types of targets like `PBXAggregateTarget`, only native ones.
     struct PBXNativeTarget: PBXObject {
-        private let isa: ISA<Self>
-
         let name: String
         let buildPhases: [ObjectUUID]
-        let productType: ProductType
-
+        let productType: String
+        var knownProductType: ProductType? { ProductType(rawValue: productType) }
+        
```

Comment:
To be honest initially I implemented this case using a Property Wrapper and I was happy with myself because it looked so beautiful… I mean look at this magic: eacf33e5c3237a062cccda06cce37d316fcf868b 🤩 

But ultimately I accepted that simpler was probably better… and said goodbye to my beloved Property Wrapper in 70f963075762f89fc2fde23d1550cc7a9ca7dde3 after all 😅